aten/src/ATen/native/quantized/cpu/qadd.cpp

```diff
@@ -9,48 +9,62 @@
 namespace at {
 namespace native {
 namespace {
+
+inline Tensor requantize_into_int32(Tensor qa) {
```

Comment:
not used?

Reply:
Also did you actually mean to take the tensor by value? Probably better to take a const ref so we don't hit atomic refcount bumps

Reply:
@jamesr66a I was told before that with the newest changes to at::Tensor, there is no need to pass it by const ref. I might have misunderstood something.

Reply:
Tensor is still a thin wrapper around intrusive_ptr<TensorImpl> which has--not one--but TWO atomic refcounts that need to be updated on copies. This is a high cost because of synchronization overheads and should be avoided if you're only reading properties from a tensor.

See my recent note about profiling with flame graphs. We've spent a lot of time recently removing unnecessary Tensor copies to reduce this overhead